ui/panels-plugin/src/plugins/time-series-chart/TimeSeriesChartPanel.tsx

```diff
@@ -11,35 +11,167 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { PanelProps, useTimeSeriesQueries } from '@perses-dev/plugin-system';
+import { useMemo, useState } from 'react';
+import { merge } from 'lodash-es';
+import { PanelProps, useTimeSeriesQueries, useTimeRange } from '@perses-dev/plugin-system';
+import { GridComponentOption } from 'echarts';
+import { Box, Skeleton } from '@mui/material';
+import { LineChart, EChartsDataFormat, ZoomEventData, Legend } from '@perses-dev/components';
 import { useSuggestedStepMs } from '../../model/time';
-import { TimeSeriesChartOptions } from './time-series-chart-model';
-import { TimeSeriesChartContainer } from './TimeSeriesChartContainer';
+import { StepOptions, ThresholdColors, ThresholdColorsPalette } from '../../model/thresholds';
+import { TimeSeriesChartOptions, DEFAULT_LEGEND } from './time-series-chart-model';
+import { getLineSeries, getCommonTimeScale, getYValues, getXValues, EMPTY_GRAPH_DATA } from './utils/data-transform';
+import { getRandomColor } from './utils/palette-gen';
 
 export type TimeSeriesChartProps = PanelProps<TimeSeriesChartOptions>;
 
 export function TimeSeriesChartPanel(props: TimeSeriesChartProps) {
   const {
-    spec: { queries, show_legend, thresholds, unit },
+    spec: { queries, thresholds },
     contentDimensions,
   } = props;
 
+  // popuate 'position' and 'show' if undefined
+  const legend = merge({}, DEFAULT_LEGEND, props.spec.legend);
+
+  const unit = props.spec.unit ?? {
+    kind: 'Decimal',
+    decimal_places: 2,
+  };
+
+  // TODO: change to array, support multi select on Shift-click
+  const [selectedSeriesName, setSelectedSeriesName] = useState<string | null>(null);
+
   const suggestedStepMs = useSuggestedStepMs(contentDimensions?.width);
   const queryResults = useTimeSeriesQueries(queries, { suggestedStepMs });
 
+  const { setTimeRange } = useTimeRange();
+
+  const onLegendItemClick = (seriesName: string) => {
+    setSelectedSeriesName((current) => {
+      if (current === null || current !== seriesName) {
+        return seriesName;
+      }
+      return null;
+    });
+  };
+
+  // populate series data based on query results
+  const { graphData, loading } = useMemo(() => {
+    const timeScale = getCommonTimeScale(queryResults);
+    if (timeScale === undefined) {
+      for (const query of queryResults) {
+        // does not show error message if any query is successful (due to timeScale check)
+        if (query.error) throw query.error;
```

Comment:
I think I'm missing something, but this reads to me that we WILL throw an error if any query is unsuccessful (even if another query is successful)

Reply:
So I did this to fix error handling awhile back in #479, but there might be a way to refactor this after all the new useQueries changes in #601 to be more intuitive 🤔 

That `if (timeScale === undefined)` line is only reason it works since [timeScale will be defined](https://github.com/perses/perses/blob/3723d62657d9cfa83e9ce00f0d26d59284873d1d/ui/panels-plugin/src/plugins/time-series-chart/utils/data-transform.ts#L42) if any single query returns successful